### PR TITLE
Social recovery validator

### PR DESCRIPTION
**Pull request by Arch0125**

### CODE REVIEW

File: packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-validator.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  });
+
+  it("should sign hash properly without 0x", async () => {
+    expect(await validator.signMessage("icanbreakthistestcase")).eql(
+      "0x4d61c5c27fb64b207cbf3bcf60d78e725659cff5f93db9a1316162117dff72aa631761619d93d4d97dfb761ba00b61f9274c6a4a76e494df644d968dd84ddcdb1c"
+    );
+  });
+
+  it(
+    "should correctly make an api call to set guardians",
```
Comment (derekchiang) [inline packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-validator.test.ts:44]: Shouldn't there also be a test for recovering?

File: packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-validator.test.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      "0x7c8999dc9a822c1f0df42023113edb4fdd543266"
+    );
+    // console.log(res);
+    expect(res).toBeDefined();
+  });
+
+  it("should generate recovery calldata correctly", async () => {
+    try {
+      const res = await validator.getrecoveryCallData(
+        globalrecoveryid,
```
Comment (derekchiang) [inline packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-validator.test.ts:128]: Small thing but please use casing correctly, like `getRecoveryCalldata` and `globalRecoveryId`

Comment (derekchiang) [reply packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-validator.test.ts:128]: Especially for public APIs

**Review by SahilVasava — CHANGES_REQUESTED** [review]

File: packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-provider.test.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        },
+      },
+    });
+
+    const data = {
+        guardians: {
+          "0x5B38Da6a701c568545dCfcB03FcB875f56beddC4": 100,
+        },
+        threshold: 50,
+        owneraddress: "0x7c8999dc9a822c1f0df42023113edb4fdd543266",
```
[the remark this answers is not in the document]
Comment (derekchiang) [reply packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-provider.test.ts:47]: Also please fix variable names like `owneraddress` to `ownerAddress`

File: packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-provider.test.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            txRetryIntervalMs: 2000,
+          },
+        },
+        paymasterConfig: {
+          policy: "VERIFYING_PAYMASTER",
+        },
+      },
+    });
+
+    const res = await socialRecoveryProvider.initRecovery(
```
Comment (derekchiang) [inline packages/accounts/src/kernel-zerodev/__tests__/kernel-recovery-provider.test.ts:75]: How come the test ends at `initRecovery`?  Aren't there a few more steps like:

- Guardians signing recovery
- Someone submitting the recovery

File: packages/accounts/src/kernel-zerodev/validator-provider/social-recovery-provider.ts [hunk 5]
```diff
@@ -9,19 +9,37 @@ import {
...
 
 export class SocialRecoveryProvider extends ValidatorProvider<SocialRecoveryValidatorParams> {
   constructor(
     params: ExtendedValidatorProviderParams<SocialRecoveryValidatorParams>
   ) {
-    const chain =
-      typeof params.opts?.providerConfig?.chain === "number"
-        ? getChain(params.opts.providerConfig.chain)
-        : params.opts?.providerConfig?.chain ?? polygonMumbai;
+    const chain = polygonMumbai;
```
Comment (SahilVasava) [inline packages/accounts/src/kernel-zerodev/validator-provider/social-recovery-provider.ts:35]: Why is this hardcoded as `polygonMumbai`?

Comment (Arch0125) [reply packages/accounts/src/kernel-zerodev/validator-provider/social-recovery-provider.ts:35]: I'll be changing this, was using it for tests

**Pull request closed without merging**